src/OrbitGl/MockBatcher.h

```diff
@@ -0,0 +1,129 @@
+
+#ifndef ORBIT_GL_MOCK_BATCHER_H_
+#define ORBIT_GL_MOCK_BATCHER_H_
+
+#include <limits>
+
+#include "BatcherInterface.h"
+
+namespace orbit_gl {
+
+class MockBatcher : public BatcherInterface {
+ public:
+  explicit MockBatcher() : BatcherInterface(BatcherId::kTimeGraph) { ResetElements(); }
+  void AddLine(Vec2 from, Vec2 to, float z, const Color& color, const Color& /*picking_color*/,
+               std::unique_ptr<PickingUserData> /*user_data*/ = nullptr) override {
+    num_lines_by_color_[color]++;
+    if (from[0] == to[0]) num_vertical_lines_++;
+    if (from[1] == to[1]) num_horizontal_lines_++;
+    ProcessPoint({from[0], from[1], z});
+    ProcessPoint({to[0], to[1], z});
+  }
+  void AddBox(const Box& box, const std::array<Color, 4>& colors, const Color& /*picking_color*/,
+              std::unique_ptr<PickingUserData> /*user_data*/ = nullptr) override {
+    num_boxes_by_color_[colors[0]]++;
+    for (int i = 0; i < 4; i++) {
+      ProcessPoint(box.vertices[i]);
+    }
+  }
+  void AddTriangle(const Triangle& triangle, const std::array<Color, 3>& colors,
+                   const Color& /*picking_color*/,
+                   std::unique_ptr<PickingUserData> /*user_data*/ = nullptr) override {
+    num_triangles_by_color_[colors[0]]++;
+    for (int i = 0; i < 3; i++) {
+      ProcessPoint(triangle.vertices[i]);
+    }
+  }
+
+  void ResetElements() override {
+    num_lines_by_color_.clear();
+    num_triangles_by_color_.clear();
+    num_boxes_by_color_.clear();
+    num_horizontal_lines_ = 0;
+    num_vertical_lines_ = 0;
+    min_point_ = Vec3{std::numeric_limits<float>::max(), std::numeric_limits<float>::max(),
+                      std::numeric_limits<float>::max()};
+    max_point_ = Vec3{std::numeric_limits<float>::lowest(), std::numeric_limits<float>::lowest(),
+                      std::numeric_limits<float>::lowest()};
+  }
+
+  [[nodiscard]] uint32_t GetNumElements() const override {
+    return GetNumLines() + GetNumBoxes() + GetNumTriangles();
+  }
+
+  [[nodiscard]] std::vector<float> GetLayers() const override { return {}; };
+  virtual void DrawLayer(float /*layer*/, bool /*picking*/) const override{};
+
+  [[nodiscard]] const PickingUserData* GetUserData(PickingId /*id*/) const override {
+    return nullptr;
+  }
+
+  [[nodiscard]] int GetNumLinesByColor(Color color) const { return num_lines_by_color_.at(color); }
+  [[nodiscard]] int GetNumTrianglesByColor(Color color) const {
+    return num_triangles_by_color_.at(color);
+  }
+  [[nodiscard]] int GetNumBoxesByColor(Color color) const { return num_boxes_by_color_.at(color); }
+  [[nodiscard]] int GetNumHorizontalLines() const { return num_horizontal_lines_; }
+  [[nodiscard]] int GetNumVerticalLines() const { return num_vertical_lines_; }
+  [[nodiscard]] bool ExpectDraw(int num_lines, int num_triangles, int num_boxes) {
+    return GetNumLines() == num_lines && GetNumTriangles() == num_triangles &&
+           GetNumBoxes() == num_boxes;
+  }
```

Comment:
What's the purpose of this function? I'm asking because I see a couple of drawbacks:

1. It takes 3 ints. It's easy to mix them up
2. Due to the name I expect this function to be plugged into the gtest EXPECT macro but it's not expecting anything on its own.
3. When plugged into EXPECT_TRUE and it fails the error message will not be great because EXPECT_TRUE won't know which of the comparisons failed. Three independent EXPECT_EQ calls are preferrable.

Reply:
I would rather return 3 different functions to after manage this expect inside the test classes. So, basically I would revert my fixup commit (where I create this method).